esrally/mechanic/supplier.py

```diff
@@ -725,7 +726,8 @@ def _update(self, revision):
 
     @classmethod
     def is_commit_hash(cls, revision):
-        return revision != "latest" and revision != "current" and not revision.startswith("@")
+        sha1_pattern = re.compile("[0-9a-f]{5,40}")
```

Comment:
I'm uncomfortable with allowing short commit hashes, if your branch is named "facefeed" for example things are going to break. https://en.wikipedia.org/wiki/Hexspeak

Maybe calling `git show-ref --heads` and looking at the exit code is more robust?

Reply:
> I'm uncomfortable with allowing short commit hashes, if your branch is named "facefeed" for example things are going to break. https://en.wikipedia.org/wiki/Hexspeak

Would you feel more comfortable if we upped the the short commit hash minimum to 11/12 (from 5)?

> Maybe calling git show-ref --heads and looking at the exit code is more robust?

Unfortunately this also doesn't account for tags either. I need to brainstorm a bit to work out what to do here.

Reply:
We discussed this offline, but I reverted the regex approach in https://github.com/elastic/rally/pull/1591/commits/ecab3d6f3a18909ae92db90ff61c9d89558e6e91, and went with the suggested `git show-ref` approach in https://github.com/elastic/rally/pull/1591/commits/5902fdff7c9fcfdebe251c542b1716093e631aaa.

From my testing https://github.com/elastic/rally/pull/1591/commits/5902fdff7c9fcfdebe251c542b1716093e631aaa allows us to properly support commit hashes, branch names, and tags.